### PR TITLE
Add Magic header for Discovery v4 udp packets

### PR DESCRIPTION
**Pull request by yazzaoui**
This adds a simple magic header to discovery packets to ensure proper isolation from the Ethereum discovery network. 
That should speed up time to find an autonity peer significantly.

### CODE REVIEW

File: p2p/discover/v4wire/v4wire.go [hunk 1]
```diff
@@ -210,8 +214,11 @@ func Decode(input []byte) (Packet, Pubkey, []byte, error) {
 	if len(input) < headSize+1 {
 		return nil, Pubkey{}, nil, ErrPacketTooSmall
 	}
-	hash, sig, sigdata := input[:macSize], input[macSize:headSize], input[headSize:]
-	shouldhash := crypto.Keccak256(input[macSize:])
+	magic, hash, sig, sigdata := input[:magicSize], input[magicSize:macSize+magicSize], input[macSize+magicSize:headSize], input[headSize:]
+	if !bytes.Equal(magic, AutMagic) {
+		return nil, Pubkey{}, nil, ErrBadHash
```
Comment (Jason-Zhangxin-Chen) [inline p2p/discover/v4wire/v4wire.go:219]: ErrBadMagic should be returned.

File: p2p/discover/v4wire/v4wire.go [hunk 2]
```diff
@@ -44,6 +44,8 @@ const (
 	ENRResponsePacket
 )
 
+var AutMagic = []byte{'a', 'u', 't'}
```
Comment (Jason-Zhangxin-Chen) [inline p2p/discover/v4wire/v4wire.go:47]: Do we need to add the `aut` magic number to the v5 discovery protocol?

**Review by vpiyush — COMMENTED** [review]
these looks good in addition to this we should probably have same checks for V5 too. We do support for v5

**Review by Jason-Zhangxin-Chen — APPROVED** [review]
LGTM now.

**Review by lorenzo-dev1 — APPROVED** [review]
LGTM

FYI I saw that discovery v5 should have something that allows to do the same without modifying the protocol (using topics, which map to the services offered by the node). See https://github.com/ethereum/devp2p/blob/master/discv5/discv5-theory.md#topic-advertisement

However I am not sure if using the ETH DHT to advertise autonity nodes would bring benefits, maybe enhanced network robustness?

**Comment by yazzaoui** [comment]
Very interesting video you linked @lorenzo-dev1. I can definitely see a benefit of sharing the DHT if Autonity was more integrated with Ethereum but that's not really the case for now and we can easily restore that possibility without hardforking.
We'll get back to this topic in the future, for now, I think we're good.

**Pull request merged**